// configure-env/dist/prompts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const prompts_1 = __importDefault(require("prompts"));
function createQuestions(parsedExample, promptStream) {
    return parsedExample.variablesToSet.map(entry => {
        return {
            type: 'text',
            name: entry.name,
            message: entry.comment,
            initial: entry.defaultValue,
            stdout: promptStream,
        };
    });
}
exports.createQuestions = createQuestions;
function handleCancel(prompt, answers) {
    /* keep asking questions */
    return true;
}
exports.handleCancel = handleCancel;
async function promptForVariables(parsedExample, promptStream) {
    const questions = createQuestions(parsedExample, promptStream);
    const answers = await prompts_1.default(questions, { onCancel: handleCancel });
    return answers;
}
exports.promptForVariables = promptForVariables;
